app/backend/src/couchers/servicers/events.py

```diff
@@ -1156,18 +1156,32 @@ def RemoveEventOrganizer(self, request, context, session):
         if occurrence.end_time < now() - timedelta(hours=24):
             context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.EVENT_CANT_UPDATE_OLD_EVENT)
 
-        if event.owner_user_id == context.user_id:
+        # Determine which user to remove
+        user_id_to_remove = request.user_id if request.HasField("user_id") else context.user_id
+
+        # Check permissions: either removing yourself OR you're the event owner
+        can_remove = (
+            context.user_id == user_id_to_remove  # Self-removal
+            or _is_event_owner(event, context.user_id)  # Event owner removing someone else
+        )
+
+        if not can_remove:
```

Comment:
This check should be

```py3
        if not _can_edit_event(session, event, context.user_id):
```

Reply:
If I understand correctly, `_can_edit_event` currently only includes event owners and moderators, but not co-organizers.

That means this change would block permission from a co-organizer removing themselves.

Reply:
Am I correct that `_can_edit_event` does NOT include co-organizers currently?

That would mean a co-organizer can no longer remove themselves once someone else makes them a co-organizer.